Error converting comma-separated values "Katalogwerte" in cleansing bulk

## Description of the issue

Running the default Mastr download() currently results in an error processing certain files - as of March 8th
- Error processing file 'Netze.xml': 'could not convert string to float: '334, 335''
- Error processing file 'EinheitenVerbrennung.xml': 'could not convert string to float: '2442, 2442''

As a result, the corresponding sqlite tables are empty.
The issue seems to be caused by the dtype check for "O" in `replace_mastr_katalogeintraege`. This check does not match the pandas string dtype introduced in pandas>=3.0

If the suggested solution outlined below aligns with your expectations, I would be happy to prepare a pull request.

Please let me know if there are any additional implications or edge cases that I may have overlooked.

## Steps to Reproduce
1. run default db.download()
2. check if grids or combustion_extended tables in sqlite are empty

## Ideas of solution
- adjust the if statement in `replace_mastr_katalogeintraege` (`utils_cleansing_bulk.py`) to include pandas string_dtype, i.e. instead of `if df[column_name].dtype == "O":`
- use `if (pd.api.types.is_string_dtype(df[column_name])
                or pd.api.types.is_object_dtype(df[column_name])):`
- as the object dtype check is still included it should work the same as before with older pandas versions

## Context and Environment
* Version used: v0.16.1 and latest commit on develop 297cd59 - specifically with pandas 3.0.1
* Operating system: unix/mac os
* Environment setup and (python) version: python 3.12, pandas 3.0.1

## Workflow checklist
- [x] I have checked [the documentation](https://open-mastr.readthedocs.io/en/latest/) and confirmed this issue is not already addressed there.
- [x] I am aware of the workflow in [CONTRIBUTING.md](https://github.com/OpenEnergyPlatform/open-MaStR/blob/production/CONTRIBUTING.md)

> I cannot reproduce this. The `grids` table got populated when I ran `python main.py` on commit `2dd7223`, yesterday around 6pm German time.

Interesting. The `grids` table works fine now with `2dd7223` as of today (2026-05-14).
Did the data change or is the error fixed with #718 ?

However, now I get other errors (same cause as mentioned above) - also `2dd7223` as of today (2026-05-14).: 

`
[ERROR] Error processing file 'EinheitenSolar_1.xml': 'could not convert string to float: '712, 812''
[ERROR] Error processing file 'EinheitenSolar_2.xml': 'could not convert string to float: '712, 3186''
[ERROR] Error processing file 'EinheitenSolar_3.xml': 'could not convert string to float: '710, 3186''
[ERROR] Error processing file 'EinheitenSolar_4.xml': 'could not convert string to float: '3181, 709''
[ERROR] Error processing file 'EinheitenSolar_5.xml': 'could not convert string to float: '812, 710''
[ERROR] Error processing file 'EinheitenSolar_6.xml': 'could not convert string to float: '812, 813''
[ERROR] Error processing file 'EinheitenSolar_7.xml': 'could not convert string to float: '812, 813''
[ERROR] Error processing file 'EinheitenSolar_8.xml': 'could not convert string to float: '813, 711''
[ERROR] Error processing file 'EinheitenSolar_9.xml': 'could not convert string to float: '712, 812''
[ERROR] Error processing file 'EinheitenSolar_10.xml': 'could not convert string to float: '812, 813''
[ERROR] Error processing file 'EinheitenSolar_11.xml': 'could not convert string to float: '710, 3182''
[ERROR] Error processing file 'EinheitenSolar_12.xml': 'could not convert string to float: '812, 813''
[ERROR] Error processing file 'EinheitenSolar_13.xml': 'could not convert string to float: '3183, 709''
[ERROR] Error processing file 'EinheitenSolar_14.xml': 'could not convert string to float: '812, 3186''
[ERROR] Error processing file 'EinheitenSolar_15.xml': 'could not convert string to float: '3186, 3181''
[ERROR] Error processing file 'EinheitenSolar_16.xml': 'could not convert string to float: '709, 710''
[ERROR] Error processing file 'EinheitenSolar_17.xml': 'could not convert string to float: '711, 3186''
[ERROR] Error processing file 'EinheitenSolar_18.xml': 'could not convert string to float: '2965, 709''
[ERROR] Error processing file 'EinheitenSolar_19.xml': 'could not convert string to float: '812, 712''
[ERROR] Error processing file 'EinheitenSolar_20.xml': 'could not convert string to float: '709, 3183''
[ERROR] Error processing file 'EinheitenSolar_21.xml': 'could not convert string to float: '711, 3183''
[ERROR] Error processing file 'EinheitenSolar_22.xml': 'could not convert string to float: '712, 3186''
[ERROR] Error processing file 'EinheitenSolar_23.xml': 'could not convert string to float: '710, 711''
[ERROR] Error processing file 'EinheitenSolar_24.xml': 'could not convert string to float: '3183, 711''
[ERROR] Error processing file 'EinheitenSolar_25.xml': 'could not convert string to float: '3183, 3186''
[ERROR] Error processing file 'EinheitenSolar_26.xml': 'could not convert string to float: '2965, 709''
[ERROR] Error processing file 'EinheitenSolar_27.xml': 'could not convert string to float: '3183, 711''
[ERROR] Error processing file 'EinheitenSolar_28.xml': 'could not convert string to float: '813, 812''
[ERROR] Error processing file 'EinheitenSolar_29.xml': 'could not convert string to float: '710, 711''
[ERROR] Error processing file 'EinheitenSolar_30.xml': 'could not convert string to float: '3183, 3186''
[ERROR] Error processing file 'EinheitenSolar_31.xml': 'could not convert string to float: '711, 3181''
[ERROR] Error processing file 'EinheitenSolar_32.xml': 'could not convert string to float: '711, 3186''
[ERROR] Error processing file 'EinheitenSolar_33.xml': 'could not convert string to float: '711, 3183''
[ERROR] Error processing file 'EinheitenSolar_34.xml': 'could not convert string to float: '712, 709''
[ERROR] Error processing file 'EinheitenSolar_35.xml': 'could not convert string to float: '710, 709''
[ERROR] Error processing file 'EinheitenSolar_36.xml': 'could not convert string to float: '711, 3186''
[ERROR] Error processing file 'EinheitenSolar_37.xml': 'could not convert string to float: '812, 712, 2965''
[ERROR] Error processing file 'EinheitenSolar_38.xml': 'could not convert string to float: '3181, 711, 709''
[ERROR] Error processing file 'EinheitenSolar_39.xml': 'could not convert string to float: '3183, 710''
[ERROR] Error processing file 'EinheitenSolar_40.xml': 'could not convert string to float: '710, 3186''
[ERROR] Error processing file 'EinheitenSolar_41.xml': 'could not convert string to float: '3057, 3183''
[ERROR] Error processing file 'EinheitenSolar_42.xml': 'could not convert string to float: '812, 813, 2965''
[ERROR] Error processing file 'EinheitenSolar_43.xml': 'could not convert string to float: '3186, 812''
[ERROR] Error processing file 'EinheitenSolar_44.xml': 'could not convert string to float: '711, 710''
[ERROR] Error processing file 'EinheitenSolar_45.xml': 'could not convert string to float: '3183, 710''
[ERROR] Error processing file 'EinheitenSolar_46.xml': 'could not convert string to float: '712, 709''
[ERROR] Error processing file 'EinheitenSolar_47.xml': 'could not convert string to float: '711, 710''
[ERROR] Error processing file 'EinheitenSolar_48.xml': 'could not convert string to float: '3181, 711''
[ERROR] Error processing file 'EinheitenSolar_49.xml': 'could not convert string to float: '712, 3181, 2965, 3180''
[ERROR] Error processing file 'EinheitenSolar_50.xml': 'could not convert string to float: '709, 710''
[ERROR] Error processing file 'EinheitenSolar_51.xml': 'could not convert string to float: '3181, 3183''
[ERROR] Error processing file 'EinheitenSolar_52.xml': 'could not convert string to float: '712, 2965''
[ERROR] Error processing file 'EinheitenSolar_53.xml': 'could not convert string to float: '710, 3182''
[ERROR] Error processing file 'EinheitenSolar_54.xml': 'could not convert string to float: '710, 711''
[ERROR] Error processing file 'EinheitenSolar_55.xml': 'could not convert string to float: '3181, 711''
[ERROR] Error processing file 'EinheitenSolar_56.xml': 'could not convert string to float: '711, 3057''
[ERROR] Error processing file 'EinheitenSolar_57.xml': 'could not convert string to float: '712, 812, 2965, 3180''
[ERROR] Error processing file 'EinheitenSolar_58.xml': 'could not convert string to float: '712, 813''
[ERROR] Error processing file 'EinheitenSolar_59.xml': 'could not convert string to float: '709, 711''
[ERROR] Error processing file 'EinheitenSolar_60.xml': 'could not convert string to float: '711, 710''
[ERROR] Error processing file 'EinheitenSolar_61.xml': 'could not convert string to float: '710, 711''
[ERROR] Error processing file 'EinheitenSolar_62.xml': 'could not convert string to float: '711, 3181''`

--- a/.bumpversion.cfg
+++ b/.bumpversion.cfg
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 0.17.1
+current_version = 0.17.2
 parse = (?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)((?P<release>(a|na))+(?P<build>\d+))?
 serialize = 
 	{major}.{minor}.{patch}{release}{build}

--- a/.github/workflows/ci-production.yml
+++ b/.github/workflows/ci-production.yml
@@ -33,7 +33,7 @@ jobs:
     - name: create package
       run: python -m build --sdist
     - name: import open-mastr
-      run: python -m pip install ./dist/open_mastr-0.17.1.tar.gz
+      run: python -m pip install ./dist/open_mastr-0.17.2.tar.gz
     - name: Create credentials file
       env:
         MASTR_TOKEN: ${{ secrets.MASTR_TOKEN }}

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,13 @@ For each version important additions, changes and removals are listed here.
 The format is inspired from [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and the versioning aims to respect [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [v0.17.2] Hotfix - 2026-07-17
+### Changed
+- Fix error where tables where empty by correctly parsing comma seperated int values
+  [#772](https://github.com/OpenEnergyPlatform/open-MaStR/pull/772)
+
+
+
 ## [v0.17.1] Hotfix - 2026-04-13
 ### Changed
 - Fix broken `Mastr.to_csv` function by removing stray `**kwargs`

--- a/CITATION.cff
+++ b/CITATION.cff
@@ -45,7 +45,7 @@ authors:
 title: "open-MaStR"
 type: software
 license: AGPL-3.0
-version: 0.17.1
+version: 0.17.2
 doi: 
-date-released: 2026-04-13
+date-released: 2026-07-17
 url: "https://github.com/OpenEnergyPlatform/open-MaStR/"

--- a/CITATION.cff
+++ b/CITATION.cff
@@ -18,7 +18,7 @@ authors:
   - family-names: "Kotthoff"
     given-names: "Florian"
     alias: "@FlorianK13"
-    affiliation: "fortiss"
+    affiliation: "OFFIS e.V. - Institute for Informatics"
     orcid: " https://orcid.org/0000-0003-3666-6122"
   - family-names: "Tepe"
     given-names: "Deniz"
@@ -42,7 +42,29 @@ authors:
     given-names: "Simon"
     alias: "@Simon-Will"
     affiliation: ""
+identifiers:
+  - type: swh
+    value: 'swh:1:dir:28252b3ca57b56d22be851bad576b6748af2e171'
+    description: Software Heritage archive identifier
 title: "open-MaStR"
+abstract: >-
+  The Python Package open-mastr provides an interface for
+  accessing and cleaning the German Energy Unit dataset
+  called Marktstammdatenregister (MaStR). The MaStR is a
+  central registry with detailed information about renewable
+  and conventional power plants in the German energy system.
+  open-mastr enables the creation and updating of a local
+  database of the entire registry, as well as processing the
+  data for further evaluation. Ultimately, the package
+  offers methods to reduce the registry's parsing time and
+  thus enables energy system researchers to start working
+  with the entire dataset right away.
+keywords:
+  - python
+  - Marktstammdatenregister
+  - Energy System
+  - Energy System Data
+  - Open Data
 type: software
 license: AGPL-3.0
 version: 0.17.2

--- a/README.rst
+++ b/README.rst
@@ -13,16 +13,14 @@ open-mastr
 .. list-table::
    :widths: 10, 50
 
-   * - License
-     - |badge_license|
-   * - Documentation
-     - |badge_rtd|
-   * - Tests
-     - |badge_ci|
+   * - General
+     - |badge_license| |badge_rtd| |badge_ci|
    * - Publication
      - |badge_pypi| |badge_joss|
    * - Data Publication
      - |badge_zenodo|
+   * - Software Archive
+     - |badge_software_heritage|
    * - Development
      - |badge_issue_open| |badge_issue_closes| |badge_pr_open| |badge_pr_closes|
    * - Community
@@ -37,7 +35,7 @@ open-mastr
 Introduction
 ============
 
-The python package ``open-mastr`` provides an interface for accessing the `Marktstammdatenregister (MaStR) <https://www.marktstammdatenregister.de/MaStR>`_ data. The MaStR is a German register provided by the German Federal Network Agency (Bundesnetzagentur / BNetza) that keeps track of all power and gas units located in Germany. It is a daily growing dataset with millions of data points covering electricity and gas production units, electricity and gas consumers, storages, grids, and energy market participants.
+The Python package ``open-mastr`` provides an interface for accessing the `Marktstammdatenregister (MaStR) <https://www.marktstammdatenregister.de/MaStR>`_ data. The MaStR is a German register provided by the German Federal Network Agency (Bundesnetzagentur / BNetza) that keeps track of all power and gas units located in Germany. It is a daily growing dataset with millions of data points covering electricity and gas production units, electricity and gas consumers, storages, grids, and energy market participants.
 
 Generally, and besides the offerings of ``open-mastr``, the MaStR data can be accessed via three main options:
 
@@ -45,7 +43,7 @@ Generally, and besides the offerings of ``open-mastr``, the MaStR data can be ac
 #. download `daily provided dumps <https://www.marktstammdatenregister.de/MaStR/Datendownload>`_
 #. access via the `web service <https://www.marktstammdatenregister.de/MaStRHilfe/subpages/webdienst.html>`_
 
-These access options, however, are not exactly frictionless. ``open-mastr`` thus provides an interface for and improved developer experience of accessing the data. This project is intended for individuals who wish to "just work" with the MaStR data and who do not want to deal with the idiosyncrasies of the three access options above.
+These access options, however, are not exactly frictionless. ``open-mastr`` thus provides an improved developer experience for accessing the data. This project is intended for individuals who wish to "just work" with the MaStR data and who do not want to deal with the idiosyncrasies of the three access options above.
 
 In particular, ``open-mastr`` facilitates access to the daily provided MaStR dumps with download methods (bulk) and by parsing the XML files to a relational database. Furthermore, the software provides a Python wrapper to access the MaStR SOAP web service (API).
 
@@ -66,6 +64,8 @@ Benefits provided by ``open-mastr``
      - Download, decode, and write data to a local database
    * - Translation to English
      - Translate table names and columns from German to English as well as an English documentation page of the dataset
+   * - Speed and parallelization
+     - Download and parse tables in seconds, not minutes
    * - Data processing
      - Merge relevant information about different technologies to single csv files
 
@@ -86,7 +86,6 @@ Installation
 ============
 
 | It is recommended to use a virtual python environment, for example `conda <https://docs.conda.io/en/latest/miniconda.html>`_ or `virtualenv <https://virtualenv.pypa.io/en/latest/installation.html>`_.
-| The package is intended to be used with ``Python >=3.8``.
 
 
 PyPI
@@ -108,17 +107,12 @@ For development, clone this repository manually.
     git clone git@github.com:OpenEnergyPlatform/open-MaStR.git
     cd open-MaStR
 
-Setup the conda environment with
-
-.. code-block:: python
-
-    conda env create -f environment.yml
 
 Install the package with
 
 .. code-block:: python
 
-    pip install "open_mastr[dev]"
+    pip install -e .[dev]
 
 
 Examples of Usage
@@ -209,4 +203,5 @@ Data
 .. |badge_joss| image:: https://joss.theoj.org/papers/dc0d33e7dc74f7233e15a7b6fe0c7a3e/status.svg
     :target: https://joss.theoj.org/papers/dc0d33e7dc74f7233e15a7b6fe0c7a3e
 
-
+.. |badge_software_heritage| image:: https://archive.softwareheritage.org/badge/swh:1:dir:28252b3ca57b56d22be851bad576b6748af2e171/
+   :target: https://archive.softwareheritage.org/swh:1:dir:28252b3ca57b56d22be851bad576b6748af2e171;origin=https://github.com/OpenEnergyPlatform/open-MaStR;visit=swh:1:snp:8065d0931f826af24c57fae612c6a48e0fcb135b;anchor=swh:1:rev:430223ddf437ba2611fe9a38521e6549ff61e750

--- a/open_mastr/xml_download/utils_cleansing_bulk.py
+++ b/open_mastr/xml_download/utils_cleansing_bulk.py
@@ -36,7 +36,7 @@ def replace_mastr_katalogeintraege(
     katalogwerte = create_katalogwerte_from_bulk_download(zipped_xml_file_path)
     for column_name in df.columns:
         if column_name in columns_replace_list:
-            if df[column_name].dtype == "O":
+            if pd.api.types.is_string_dtype(df[column_name]):
                 # Handle comma seperated strings from catalog values
                 df[column_name] = (
                     df[column_name]

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "open_mastr"
-version = "0.17.1"
+version = "0.17.2"
 dependencies = [
   "pandas>=2.2.2",
   "numpy",
@@ -79,4 +79,4 @@ open_mastr = [
 include = ["open_mastr", "open_mastr.soap_api", "open_mastr.soap_api.metadata", "open_mastr.utils", "open_mastr.utils.config", "open_mastr.xml_download"] # package names should match these glob patterns (["*"] by default)
 
 # from setup.py - not yet included in here
-# download_url="https://github.com/OpenEnergyPlatform/open-MaStR/archive""/refs/tags/v0.17.1.tar.gz",
+# download_url="https://github.com/OpenEnergyPlatform/open-MaStR/archive""/refs/tags/v0.17.2.tar.gz",

